README.md

```diff
@@ -1,32 +1,254 @@
-code synchronizer:
+### Process overview for generating a client library
 
-1. Invoking directly through a gradle task (see the build.gradle:runSynchronizer
-   task).
-2. Invoking a code generation pipeline which contains synchronization step
-   through [artman](https://github.com/googleapis/artman/#usage).
+1. Generate a descriptor file from the proto (once per API)
+2. Run client config generation (once per API)
+3. Manually tweak the generated client config
+4. Run code generation (once per API + language)
+
+Each of these steps are described in more detail below.
+
+### Set up prerequisites for building from source and running directly
+
+First, you need to install protoc if you don't have it yet: see
+[protocol-compiler-installation](https://github.com/google/protobuf#protocol-compiler-installation).
+The pre-built binaries are the easiest, but there aren't really instructions for installation -
+you have to just download the correct binaries for your platform at put them on your path.
+If you are using Linux x86_64:
+
+1. Download the latest zip from https://github.com/google/protobuf/releases ending in "-linux-x86_64.zip"
+2. Unzip the file, e.g. `unzip protoc-3.5.1-linux-x86_64.zip -d protoc-3.5.1`
+3. Copy protoc to a location on $PATH, e.g. `cp protoc-3.5.1/bin/protoc /usr/local/bin`
+
+Clone the present repository and build API Client Generator:
+
+```
+git clone https://github.com/googleapis/toolkit.git
+cd toolkit
+./gradlew fatJar
+```
+
+The googleapis/toolkit directory will hereafter be referenced as `${TOOLKIT_DIR}`.
+
+Clone the googleapis/googleapis repository (it has some config that is needed):
+
+```
+git clone https://github.com/googleapis/googleapis
+```
+
+The googleapis/googleapis directory will hereafter be referenced as `${GOOGLEAPIS_DIR}`.
+
+### Generate a descriptor file from the proto
+
+You need to locate/decide on the following before you can generate the descriptor file:
+
+1. The include directory with the proto files bundled with protoc (from the protoc setup step).
+   Hereafter, this will be referenced as `${PROTOC_INCLUDE_DIR}`.
+2. Any directories containing protos that your API depends on. For Google APIs defined in
+   [googleapis](https://github.com/googleapis/googleapis), this will be `${GOOGLEAPIS_DIR}`.
+3. The directory containing your proto. Hereafter, this will be referenced as `${YOUR_PROTO_DIR}`.
+4. Your proto file. Hereafter, this will be referenced as `${YOUR_PROTO_FILE}`.
+4. The output file name to contain the descriptor file. Hereafter, this will be referenced as
+   `${YOUR_DESCRIPTOR_FILE}`.
+
+Run the following command to generate the proto descriptor file:
+
+```
+protoc -I=${PROTOC_INCLUDE_DIR} -I=${GOOGLEAPIS_DIR} -I=${YOUR_PROTO_DIR} \
+  --include_imports --include_source_info -o ${YOUR_DESCRIPTOR_FILE} ${YOUR_PROTO_FILE}
+```
+
+### Generate proto message classes
+
+(Skip this section for Node.js - it loads proto files into memory at runtime.)
```

Comment:
Please highlight this Node.js discrepancy somehow (very easy to miss, I actually missed during first read).

Reply:
Ok I will italicize it, and also add a reminder in step `#2`.